addons/purchase/tests/test_purchase.py

```diff
@@ -942,3 +942,40 @@ def test_vendor_price_by_purchase_order_company(self):
         self.assertEqual(po.amount_untaxed, 15.0)
         po.company_id = company_a.id
         self.assertEqual(po.amount_untaxed, 10.0)
+
+    def test_bill_in_purchase_matching_individual(self):
+        """
+        Tests that if the vendor is an individual with a company, the bill will still appear when
+        using the purchase matching button on a vendor bill
+        """
+        company_partner = self.env['res.partner'].create({
+            'name': 'Small Company',
+            'company_type': 'company',
+        })
+        self.partner_a.parent_id = company_partner
+        purchase_order = self.env['purchase.order'].create({
+            'partner_id': self.partner_a.id,
+            'order_line': [(0, 0, {
+                'product_id': self.product_a.id,
+                'product_qty': 1,
+                'price_unit': 100,
+            })]
+        })
+        purchase_order.button_confirm()
+        vendor_bill = self.env['account.move'].create({
+            'move_type': 'in_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [(0, 0, {
+                'product_id': self.product_a.id,
+                'quantity': 1,
+                'price_unit': 100,
+            })]
+        })
+        self.env.flush_all()
```

Comment:
I'm not a big fan of flushing everything
Here, considering the SQL query of `purchase.bill.line.match`, flushing AML/POL will probably be enough